TextureHolder.cpp

```diff
@@ -0,0 +1,45 @@
+#include "TextureHolder.h"
+#include <assert.h>
+
+using namespace sf;
+
+TextureHolder* TextureHolder::m_s_Instance = NULL;
+
+TextureHolder::TextureHolder()
+{
+	assert(m_s_Instance == NULL);
+	m_s_Instance = this;
+}
+
+
+Texture& TextureHolder::GetTexture(std::string const& filename)
+{
+	//Get a reference to m_Textures using m_s_Instance
+	auto& m = m_s_Instance->m_Textures;

```

Comment:
`m` is not the best of names, I would recommend just naming this `textures`. Makes it easier to read!